### PR TITLE
Feature/lpfg 130 retrieve and update civil servant details

### PR DESCRIPTION
**Pull request by smoga-git**
Okay, this pulls in details from identity service and registry to create a user profile. User profile can be edited and the changes saved to the registry

**Jen's comments on branch**

Comment on #269 Feature/lpfg-288
branch LPG-130-Retrieve-and-Update-Civil-Servant-Details has now rebased onto this and any further changes will be done there. :wave:

*The changes made in this branch include*

• Querying Registry Service for professions and job roles
• Helper functions to parse the traverson data
• Logic to show the levels
• On the profession/job role levels selection page, when selecting a user it will append to the url `/${selection}`. `${selection}` maps directly to an object from the response
   • The object has three properties `self`, `name` (and `hasChildren` if it is a jobRole)
   • The `self` is a url to this object on the Registry Service and it is used to query the next call, using `.from(${self}).follow(${children})`.
   • If `hasChildren` is false, there is no call to it's children, and instead on the selection page, a query is appended to the url. `select=${profession || jobRole}/${object.id}`. This is used to update the user.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,15 +47,16 @@
...
 		"nodemon": "1.14.12",
 		"protobufjs": "6.8.4",
 		"slugid": "1.1.0",
+		"traverson-hal": "^6.0.0",
 		"tslint": "5.9.1",
 		"typescript": "2.7.1"
 	},
 	"dependencies": {
 		"@types/express-validator": "3.0.0",
+		"@types/traverson": "^2.0.28",
```
Comment (jenbutongit) [inline package.json]: Please use exact versions

File: package.json [hunk 2]
```diff
@@ -91,6 +93,8 @@
 		"striptags": "3.1.1",
 		"svelte": "1.55.0",
 		"tmp": "0.0.33",
+		"traverson-hal": "^6.0.0",
+		"traverson-promise": "0.0.9",
```
Comment (jenbutongit) [inline package.json]: If you decided not to use traverson-promise please remove

File: src/lib/config/index.ts [hunk 3]
```diff
@@ -142,3 +142,6 @@ export const XAPI = set({
 })
 
 export const YOUTUBE_API_KEY = env.YOUTUBE_API_KEY
+
+export const REGISTRY_SERVICE_URL =
+	env.REGISTRY_SERVICEURL || 'http://localhost:9002'
```
Comment (jenbutongit) [inline src/lib/config/index.ts]: REGISTRY_SERVICE_URL?

Comment (superafroman) [reply src/lib/config/index.ts]: Missing an underscore :)

File: src/lib/config/passport.ts [hunk 4]
```diff
@@ -104,8 +108,8 @@ export function hasRole(role: string) {
...
 
 export function logout(req: express.Request, res: express.Response) {
-	strategy.logout(req, (err, url) => {
-		req.logout()
-		res.redirect(url)
-	})
+	// strategy.logout(req, (err, url) => {
+	// 	req.logout()
+	// 	res.redirect(url)
+	// })
```
Comment (superafroman) [inline src/lib/config/passport.ts]: What's the deal here? No longer supported? If there's no need to do a logout on the strategy can we still do a `req.logout()` and `res.redirect(url)`?

**Pull request merged**